components/locid/src/locale.rs

```diff
@@ -151,6 +152,50 @@ impl Locale {
     ) -> Option<&extensions::unicode::Value> {
         self.extensions.unicode.keywords.get(key)
     }
+
+    /// Compare this `Locale` with a BCP-47 string.
+    ///
+    /// The return value is equivalent to what would happen if you first converted this
+    /// `Locale` to a BCP-47 string and then performed a byte comparison.
+    ///
+    /// This function is case-sensitive and results in a *total order*, so it is appropriate for
+    /// binary search. The only argument producing [`Ordering::Equal`] is `self.to_string()`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::Locale;
+    /// use std::cmp::Ordering;
+    ///
+    /// let bcp47_strings: &[&[u8]] = &[
+    ///     b"pl-Latn-PL",
+    ///     b"und",
+    ///     b"und-fonipa",
+    ///     b"und-t-m0-true",
+    ///     b"und-u-ca-hebrew",
+    ///     b"und-u-ca-japanese",
+    ///     b"zh",
+    /// ];
+    ///
+    /// for ab in bcp47_strings.windows(2) {
+    ///     let a = ab[0];
+    ///     let b = ab[1];
+    ///     assert!(a.cmp(b) == Ordering::Less);
+    ///     let a_langid = Locale::from_bytes(a).unwrap();
+    ///     assert!(a_langid.cmp_bytes(b) == Ordering::Less);
+    /// }
+    /// ```
+    pub fn cmp_bytes(&self, other: &[u8]) -> Ordering {
+        let base_iter = self.iter_subtags().map(str::as_bytes);
+        // Note: This does not use get_subtag_iterator because we want to guarantee
+        // perfect lexicographic ordering of the strings.
+        let other_iter = other.split(|b| *b == b'-');
```

Comment:
question: does the `-` here actually matter? Iterator::cmp will compare element by element up til the first unequal element, and `-` sorts lower than alphanumerics so "end of element" in an iterator is the same as a `-` in the string when comparing

Reply:
This is based on comparing subtags to subtags. I split the string so I get a subtag iterator for the string, which I compare to the subtag iterator from the structured Locale type. I'm not sure I understand your suggested action? Should I switch to using `split_inclusive`?

Reply:
oh. nevermind, misunderstood a part of the code